src/sessions/RevocationModal.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { useState } from "react";
+import {
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  Form,
+  FormGroup,
+  Modal,
+  ModalVariant,
+  TextContent,
+  TextInput,
+  ValidatedOptions,
+} from "@patternfly/react-core";
+import { useTranslation } from "react-i18next";
+import { useForm } from "react-hook-form";
+
+import type UserRepresentation from "keycloak-admin/lib/defs/userRepresentation";
+import { emailRegexPattern } from "../util";
+import type RealmRepresentation from "keycloak-admin/lib/defs/realmRepresentation";
+import { useAdminClient, useFetch } from "../context/auth/AdminClient";
+import { useRealm } from "../context/realm-context/RealmContext";
+import type ClientRepresentation from "keycloak-admin/lib/defs/clientRepresentation";
+import { useAlerts } from "../components/alert/Alerts";
+import type GlobalRequestResult from "keycloak-admin/lib/defs/globalRequestResult";
+
+type RevocationModalProps = {
+  handleModalToggle: () => void;
+  activeClients: ClientRepresentation[];
+  save: (realm?: UserRepresentation) => void;
```

Comment:
I don't see `realm` being passed or used anywhere, so I think this could be removed.
```suggestion
  save: () => void;
```

Reply:
nice catch